### PR TITLE
Add ability to set dataset project id

### PR DESCRIPTION
**Pull request by mmotoyama**
This CL adds the ability to specify a dataset project id as part of the dataset field in the connection string used to create a
BQConnection instance. Previously, we used the project defined in the URL's path component for both the billing and default dataset projects. Recall that the default project and dataset are used to disambiguate unqualified table names referenced in a query being processed by a connection.

Now, we parse the potentially fully qualified dataset to check for a project id. If no project id is found, we revert back to our former behavior, using the billing project as the default dataset project.

### CODE REVIEW

File: src/test/java/net/starschema/clouddb/jdbc/PreparedStatementTests.java
Comment (mmotoyama) [inline src/test/java/net/starschema/clouddb/jdbc/PreparedStatementTests.java:1]: Note that the modifications in this file are unrelated to the changes made to support dataset project id. The updates I made to this file were necessary for the test suite to pass.

**Review by nestorenko — APPROVED** [review]

File: src/main/java/net/starschema/clouddb/jdbc/BQSupportFuncts.java [hunk 1]
```diff
@@ -692,7 +699,8 @@ static Bigquery.Jobs.Query getSyncQuery(
       qr = qr.setJobCreationMode(jobCreationMode.name());
     }
     if (dataSet != null) {
-      qr.setDefaultDataset(new DatasetReference().setDatasetId(dataSet).setProjectId(projectId));
+      qr.setDefaultDataset(
+          new DatasetReference().setDatasetId(dataSet).setProjectId(dataSetProjectId));
```
[the remark this answers is not in the document]
Comment (mmotoyama) [reply src/main/java/net/starschema/clouddb/jdbc/BQSupportFuncts.java:703]: That is correct! We now pass the `defaultProjectId` to `configureDataSet` in `BQConnection`, so I moved the default value check for `datasetProjectId` from `parseDatasetRef` to `configureDataSet`. I removed the `defaultProjectId` parameter from `parseDatasetRef`, and I now set the project id field to `null` in the returned `DatasetReference` if no value exists in the dataset expression.
